pkg/build/api/types.go

```diff
@@ -842,7 +843,6 @@ type JenkinsPipelineBuildStrategy struct {
 	Jenkinsfile string
 
 	// Env contains additional environment variables you want to pass into a build pipeline.
-	// ValueFrom is not supported.
```

Comment:
since based on other discussion i think we're supporting ValueFrom in BuildRequest.Env now too, please update the doc there too (And make sure our validation allows it).